packages/clerk-js/src/ui/components/SignIn/SignInSocialButtons.tsx

```diff
@@ -19,11 +19,30 @@ export const SignInSocialButtons = React.memo((props: SocialButtonsProps) => {
   const signIn = useCoreSignIn();
   const redirectUrl = buildSSOCallbackURL(ctx, displayConfig.signInUrl);
   const redirectUrlComplete = ctx.afterSignInUrl || '/';
+  const shouldUsePopup = ctx.oauthFlow === 'popup' || (ctx.oauthFlow === 'auto' && originPrefersPopup());
 
   return (
     <SocialButtons
       {...props}
+      idleAfterDelay={!shouldUsePopup}
       oauthCallback={strategy => {
+        if (shouldUsePopup) {
+          // We create the popup window here with the `about:blank` URL since some browsers will block popups that are
+          // opened within async functions. The `signInWithPopup` method handles setting the URL of the popup.
+          const popup = window.open('about:blank', '', 'width=600,height=800');
+          // Unfortunately, there's no good way to detect when the popup is closed, so we simply poll and check if it's closed.
+          const interval = setInterval(() => {
+            if (popup?.closed) {
```

Comment:
The return value from `window.open` could potentially be `null` so we might want to handle that (https://developer.mozilla.org/en-US/docs/Web/API/Window/open#return_value)

```suggestion
            if (!popup || popup.closed) {
```